astra/src/test/java/com/slack/astra/server/ManagerApiGrpcTest.java

```diff
@@ -452,29 +460,805 @@ public void shouldUpdatePartitionAssignments() {
+
+    int existingDatasetThroughputBytes = 1_000;
+    List<String> existingPartitions = List.of("1", "2");
+    datasetMetadataStore.createAsync(
+        new DatasetMetadata(
+            "existingDataset1",
+            "existingDatasetOwner",
+            existingDatasetThroughputBytes,
+            List.of(
+                new DatasetPartitionMetadata(
+                    Instant.now().toEpochMilli(), MAX_TIME, existingPartitions)),
+            "whatever"));
+    createPartitions(existingPartitions, existingDatasetThroughputBytes / 2);
+
+    String datasetName = "testDataset";
+    createEmptyDatasetGRPC(datasetName, "testOwner");
+
+    long nowMs = Instant.now().toEpochMilli();
+    long throughputBytes = 1_000;
+    ManagerApi.UpdatePartitionAssignmentResponse updateResponse =
+        updatePartitionAssignmentGRPC(datasetName, throughputBytes);
+    assertThat(updateResponse.getAssignedPartitionIdsList()).isEqualTo(List.of("1", "2"));
+    await().until(() -> datasetHasPartitionConfigAfterTime(datasetName, nowMs));
+
+    assertThat(
+            latestPartitionConfig(datasetMetadataStore.getSync(datasetName)).getPartitions().size())
+        .isEqualTo(2);
+    assertThat(getPartitionMetadata("1", "2"))
+        .are(
+            sharedPartitionsWithCapacity(throughputBytes / 2 + existingDatasetThroughputBytes / 2));
+  }
+
+  @Test
+  public void shouldAutoAssignAddShared3() {
+    // 3. 1 existing shared dataset, 3 partitions, 2 used, one free, but not enough space without
+    // reassigning
+
+    // existing using max capacity * 2 - 1 on two partitions
+    // requesting max capacity * 1 - 1
+    // would require the existing dataset to be reassigned to all three partitions
+    int existingDatasetThroughputBytes = DEFAULT_MAX_CAPACITY * 2 - 2;
+    List<String> usedExistingPartitions = List.of("1", "2");
+    datasetMetadataStore.createAsync(
+        new DatasetMetadata(
+            "existingDataset1",
+            "existingDatasetOwner",
+            existingDatasetThroughputBytes,
+            List.of(
+                new DatasetPartitionMetadata(
+                    Instant.now().toEpochMilli(), MAX_TIME, usedExistingPartitions)),
+            "whatever"));
+    createPartitions(usedExistingPartitions, existingDatasetThroughputBytes / 2);
+    createPartitions("3");
+
+    String datasetName = "testDataset";
+    createEmptyDatasetGRPC(datasetName, "testOwner");
+
+    long throughputBytes = DEFAULT_MAX_CAPACITY;
+    assertThatThrownBy(() -> updatePartitionAssignmentGRPC(datasetName, throughputBytes))
+        .isInstanceOfSatisfying(
+            StatusRuntimeException.class,
+            withGrpcStatusAndDescription(
+                Status.FAILED_PRECONDITION,
+                "Needed 2 partitions with enough capacity, found 1: [3]"));
+  }
+
+  @Test
+  public void shouldAutoAssignAddShared4() {
+    // 4. 1 existing shared dataset, 3 partitions, 3 used, enough space if assigned to 3 but not
```

Comment:
This unit test case looks same as the previous one and that test failed. Let's make sure this test is correct since this one passes.